### PR TITLE
Enforce doc comments

### PR DESCRIPTION
**Pull request by idky137**
This PR will create clippy warnings for any pub functions that do not have Doc comments.

All current pub functions that do not already have a doc comment will be given "/// TODO: Add Doc Comment Here!". This way the code base will build without new warnings, but all new functions should be properly documented.

### CODE REVIEW

**Comment by fluidvanadium** [comment]
can we try this against stable

File: zingo-netutils/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,7 @@
+//! Zingo-Netutils
```
Comment (zancas) [inline zingo-netutils/src/lib.rs:1]: @AloeareV can you provide a more information doc-comment here?

File: zingo-status/src/confirmation_status.rs [hunk 2]
```diff
@@ -1,3 +1,5 @@
+//! TODO: Add Mod Description Here!
```
Comment (zancas) [inline zingo-status/src/confirmation_status.rs:1]: @fluidvanadium can you make this doc-comment more useful?

File: zingo-status/src/lib.rs [hunk 3]
```diff
@@ -1,4 +1,6 @@
-#[forbid(unsafe_code)]
+//! Zingo-Status
+//! TODO: Add Crate Discription Here!
```
Comment (zancas) [inline zingo-status/src/lib.rs:2]: @fluidvanadium can you please add a comment here?

**Comment by idky137** [comment]
Finished adding todos. There is still one warning/error generated. This is because a function is being generated in the build process, I believe the way to solve this is to add "#[allow(missing_docs)]" to the module that is generating this function. However I am unaware of where this is.

File: zingo-memo/src/lib.rs [hunk 4]
```diff
@@ -1,3 +1,9 @@
+//! Zingo-Memo
+//! Utilities for procedural creation and parsing of the Memo field
+//! These memos will likely never be directly exposed to the user,
```
Comment (zancas) [inline zingo-memo/src/lib.rs]: I prefer to remove this speculative statement.

File: zingo-memo/src/lib.rs [hunk 5]
```diff
@@ -1,3 +1,9 @@
...
+//! but instead write down things on-chain, for recovery after rescan, etc
```
Comment (zancas) [inline zingo-memo/src/lib.rs]: It's not clear to me that things that are procedurally generated/parsed by zingo-memo will always be mutually exclusive with end-user relevant content.   Perhaps we should remove this line, or refine it to be clearer?

File: zingo-netutils/src/lib.rs (hunk 1 above)
Comment (zancas) [reply zingo-netutils/src/lib.rs:1]: Are all of the current net-utils exclusively for test?

**Review by zancas — CHANGES_REQUESTED** [review]

File: zingo-memo/src/lib.rs (hunk 4 above)
Comment (AloeareV) [reply zingo-memo/src/lib.rs]: Why?

File: zingo-memo/src/lib.rs (hunk 5 above)
Comment (AloeareV) [reply zingo-memo/src/lib.rs]: > It's not clear to me that things that are procedurally generated/parsed by zingo-memo will always be mutually exclusive with end-user relevant content. Perhaps we should remove this line, or refine it to be clearer?

It's presumptive...but we can always re-evaluate it if we come up with a use-case. I can reword it to be a 'currently'.

Comment (AloeareV) [reply zingo-memo/src/lib.rs]: Reworded.

File: zingoconfig/src/lib.rs [hunk 6]
```diff
@@ -1,4 +1,8 @@
+//! ZingConfig
```
Comment (zancas) [inline zingoconfig/src/lib.rs:1]: @Oscar-Pepper can you please add a description of what this crate is, here?    No need for perfection, just enough to give a reader a general idea.  Fine to include `TODO`s for enhancements to the doc-comment.

File: zingolib/src/wallet.rs [hunk 7]
```diff
@@ -190,23 +218,23 @@ pub struct LightWallet {
...
+    /// Progress of an outgoing transaction
     send_progress: Arc<RwLock<SendProgress>>,
 
-    // The current price of ZEC. (time_fetched, price in USD)
+    /// The current price of ZEC. (time_fetched, price in USD)
     pub price: Arc<RwLock<WalletZecPriceInfo>>,
 
-    // Local state needed to submit [compact]block-requests to the proxy
-    // and interpret responses
+    /// Local state needed to submit [compact]block-requests to the proxy
```
Comment (AloeareV) [inline zingolib/src/wallet.rs]: Square brackets have special meaning in doc comments as intra-doc links, which causes this to fail cargo doc, as this is a broken link.

**Review by AloeareV — DISMISSED** [review]

File: zingolib/src/wallet.rs (hunk 7 above)
Comment (AloeareV) [reply zingolib/src/wallet.rs]: Oh hang on I will just fix this, that's easier than leaving a comment.

**Comment by fluidvanadium** [comment]
how do i run this doc-comment completeness test locally?

**Review by zancas — DISMISSED** [review]

**Review by zancas — DISMISSED** [review]

**Review by zancas — DISMISSED** [review]

**Review by zancas — APPROVED** [review]
